doc/ox-hugo-manual.org

```diff
@@ -515,39 +516,61 @@ Enable `org-hugo-export-subtree-to-md-after-save'."
-#+END_SRC
-**** Steps that /might/ need to be taken every time
-***** Step 3: Start the engines (Hugo Server)
-We start the =hugo server= so that we can see the live-preview each
-time the Org file is saved.
-
-I recommend using Hugo version 0.25 at the minimum as that added
-support for the awesome =--navigateToChanged= switch!
-
-Run below in your Hugo site root (the directory that contains the site
-=config.toml=) to start the server:
-#+BEGIN_EXAMPLE
+ #+END_SRC
+***** Steps that /might/ need to be taken every time
+****** Step 3: Start the engines (Hugo Server)
+ We start the =hugo server= so that we can see the live-preview each
+ time the Org file is saved.
+
+ I recommend using Hugo version 0.25 at the minimum as that added
+ support for the awesome =--navigateToChanged= switch!
+
+ Run below in your Hugo site root (the directory that contains the site
+ =config.toml=) to start the server:
+ #+BEGIN_EXAMPLE
 hugo server -D --navigateToChanged
-#+END_EXAMPLE
-***** Step 4: Open your browser
-By default the site is served locally on port /1313/ on
-/localhost/. So the above step would have printed something like below
-at the end:
-#+BEGIN_EXAMPLE
+ #+END_EXAMPLE
+****** Step 4: Open your browser
+ By default the site is served locally on port /1313/ on
+ /localhost/. So the above step would have printed something like below
+ at the end:
+ #+BEGIN_EXAMPLE
 Web Server is available at http://localhost:1313/ (bind address 127.0.0.1)
-#+END_EXAMPLE
+ #+END_EXAMPLE
 
-So open your favorite browser pointing to that address.
-**** FINAL step that needs to be taken every time
-If you are like me, you might not need to repeat steps 3 and 4 above,
-as you can leave the =hugo= server running in a separate terminal, and
-have a browser tab pinned to that localhost.
+ So open your favorite browser pointing to that address.
+***** FINAL step that needs to be taken every time
+ If you are like me, you might not need to repeat steps 3 and 4 above,
+ as you can leave the =hugo= server running in a separate terminal, and
+ have a browser tab pinned to that localhost.
 
-So with that, have the emacs and browser frames set up side-by-side,
-and edit your Org post.
+ So with that, have the emacs and browser frames set up side-by-side,
+ and edit your Org post.
 
-Hit =C-x C-s= and be in awe as the browser auto-refreshes to the
-*exact post you modified*!
+ Hit =C-x C-s= and be in awe as the browser auto-refreshes to the
+ *exact post you modified*!
+**** One post per Org file
+If you use a seperate Org file for each blog post, you can add this
+hook to your config:
+
+#+BEGIN_SRC emacs-lisp
+   (use-package ox-hugo
+     :ensure t
+     :after ox
+     :init
+     ; Whenever I save the Org source file for one of my blog posts,
+     ; auto-export it to hugo with ox-hugo. (Only applicable to Org
+     ; files saved in my hugo src folder in ~/git/blog/src)
+     (add-hook 'org-mode-hook
+         (lambda ()
+             (add-hook 'after-save-hook (lambda () 
```

Comment:
- Can you please separate the bare lambda into a different function? 
- And may be put an extra check that the file contains a `#+TITLE`? Because in the one Org file per post flow, that must be always be there, right?

As I realized, it is possible for a single site to use both: per file and per subtree flows.. so instead of blindly running `org-hugo-export-to-md`, might be better to check if the file looks like it is designed to export just one post.